Assets/PatchKit Patcher/Scripts/AppUpdater/AppUpdaterStrategyResolver.cs

```diff
@@ -14,7 +15,37 @@ public AppUpdaterStrategyResolver()
             DebugLogger.LogConstructor();
         }
 
-        public IAppUpdaterStrategy Resolve(AppUpdaterContext context)
+        public IAppUpdaterStrategy Create( StrategyType type, AppUpdaterContext context )
+        {
+            switch ( type )
+            {
+                case StrategyType.Empty:
+                    return new AppUpdaterEmptyStrategy();
+                case StrategyType.Content:
+                    return new AppUpdaterContentStrategy( context );
+                case StrategyType.Diff:
+                    return new AppUpdaterDiffStrategy( context );
+                default:
+                    return new AppUpdaterContentStrategy( context );
+            }
+        }
+
+        public StrategyType GetFallbackStrategy( StrategyType strategyType )
+        {
+            switch ( strategyType )
+            {
+                case StrategyType.Empty:
+                    return StrategyType.Empty;
+                case StrategyType.Content:
```

Comment:
What actually is fallback from Content? Content once again?

Reply:
If the content has been successfully downloaded and there was an error during unpacking procedure, there's no more fallback possibilities. Of course error may be caused by user - for instance it can be low disk space. A retry button should appear and try again with "content" staregy.

Reply:
I think that we should change the return type to `StrategyType?` and return `null` in case of trying to get fallback type for `StrategyType.Content`.

Then it should be also handled appropriately in AppUpdater::Update - rethrowing an exception if there's no fallback type.

Reply:
Tried avoid passing nulls around. Handling it with StrategyType.None will be ok?

Reply:
Yeah! ;-)